sqlite3.go

```diff
@@ -137,6 +137,61 @@ _sqlite3_prepare_v2_internal(sqlite3 *db, const char *zSql, int nBytes, sqlite3_
 }
 #endif
 
+static int _sqlite3_prepare_v2(sqlite3 *db, const char *zSql, int nBytes, sqlite3_stmt **ppStmt, int *oBytes) {
+	const char *tail;
+	int rv = _sqlite3_prepare_v2_internal(db, zSql, nBytes, ppStmt, &tail);
+	if (rv != SQLITE_OK) {
+		return rv;
+	}
+	if (tail) {
+		// Set oBytes to the number of bytes consumed instead of using the
+		// **pzTail out param since that requires storing a Go pointer in
+		// a C pointer, which is not allowed by CGO and will cause
+		// runtime.cgoCheckPointer to fail.
+		*oBytes = tail - zSql;
+	} else {
+		// NB: this should not happen, but if it does advance oBytes to the
+		// end of the string so that we do not loop infinitely.
+		*oBytes = nBytes;
+	}
+	return SQLITE_OK;
+}
+
+// _sqlite3_exec_no_args executes all of the statements in zSql. None of the
+// statements are allowed to have positional arguments.
+int _sqlite3_exec_no_args(sqlite3 *db, const char *zSql, int nBytes, int64_t *rowid, int64_t *changes) {
+	while (*zSql && nBytes > 0) {
+		sqlite3_stmt *stmt;
+		const char *tail;
+		int rv = sqlite3_prepare_v2(db, zSql, nBytes, &stmt, &tail);
+		if (rv != SQLITE_OK) {
+			return rv;
+		}
+
+		// Process statement
+		do {
+			rv = _sqlite3_step_internal(stmt);
+		} while (rv == SQLITE_ROW);
+
+		// Only record the number of changes made by the last statement.
+		*changes = sqlite3_changes64(db);
+		*rowid = sqlite3_last_insert_rowid(db);
+
+		if (rv != SQLITE_OK && rv != SQLITE_DONE) {
+			sqlite3_finalize(stmt);
+			return rv;
+		}
+		rv = sqlite3_finalize(stmt);
+		if (rv != SQLITE_OK) {
+			return rv;
+		}
+
+		nBytes -= tail - zSql;
```

Comment:
Here you are assuming `tail != NULL` but above in `_sqlite3_prepare_v2` you have explicit handling for that. I think we should be consistent.